site/app/controllers/admin/ConfigurationController.php

```diff
@@ -112,6 +113,13 @@ public function updateConfiguration() {
             $entry = nl2br($entry);
         }
 
+        if($name === 'forum_enabled') {
+            if($entry == 1){
+                $ForumController = new ForumController($this->core);
```

Comment:
This function makes it safe to insert the category, performing checks before doing so. It might lead to redundant code which has to be changed in two places if ever the method of entry of categories changes.